crates/datasources/src/object_store/glob_util.rs

```diff
@@ -0,0 +1,155 @@
+use std::vec::IntoIter;
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+use tracing::debug;
+
+static VALID_STRING_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new("^[a-zA-Z0-9_-]+$").unwrap());
+
+#[derive(Debug, Clone)]
+struct Replacer {
+    start: usize,
+    end: usize,
+    iter: IntoIter<String>,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Replacer find error: {0}")]
+struct ReplacerFindError(String);
```

Comment:
should we use an enum/this error?

Reply:
Any benefits of doing that?

Reply:
just that we do enums in lots of places. it's fine if that makes sense.

Reply:
We just want the error for debug log. All errors here are ignored, so didn't seem useful to have an enum. It's simply an error message.